src/main/java/net/sf/jabref/gui/journals/ManageJournalAbbreviationsViewModel.java

```diff
@@ -0,0 +1,282 @@
+import net.sf.jabref.gui.BasePanel;
+import net.sf.jabref.logic.journals.Abbreviation;
+
+import javafx.beans.property.SimpleIntegerProperty;
+import javafx.beans.property.SimpleListProperty;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.beans.property.SimpleStringProperty;
+import javafx.collections.FXCollections;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class ManageJournalAbbreviationsViewModel {
+
+    private final Log logger = LogFactory.getLog(ManageJournalAbbreviationsViewModel.class);
+    private final SimpleListProperty<AbbreviationsFile> journalFiles = new SimpleListProperty<>(
+            FXCollections.observableArrayList());
+    private final SimpleListProperty<Abbreviation> abbreviations = new SimpleListProperty<>(
+            FXCollections.observableArrayList());
+    private final SimpleIntegerProperty abbreviationsCount = new SimpleIntegerProperty();
+    private final SimpleStringProperty abbreviationsName = new SimpleStringProperty();
+    private final SimpleStringProperty abbreviationsAbbreviation = new SimpleStringProperty();
+    private final SimpleObjectProperty<AbbreviationsFile> currentFile = new SimpleObjectProperty<>();
+    private final SimpleObjectProperty<Abbreviation> currentAbbreviation = new SimpleObjectProperty<>();
+
+
+    public ManageJournalAbbreviationsViewModel() {
+        abbreviationsCount.bind(abbreviations.sizeProperty());
+    }
+
+    /**
+     * Read all saved file paths and read their abbreviations
+     */
+    public void createFileObjects() {
+        List<String> externalFiles = Globals.prefs.getStringList(JabRefPreferences.EXTERNAL_JOURNAL_LISTS);
+        externalFiles.forEach(name -> {
+            try {
+                openFile(new File(name));
+            } catch (JabRefException e) {
+                logger.debug(e);
+            }
+        });
+    }
+
+    /**
+     * This method shall be used to add a new journal abbreviation file to the
+     * set of journal abbreviation files. It basically just calls the
+     * {@link #openFile(File file)} method
+     *
+     * @param file object to be created
+     * @throws JabRefException if journal file with same path is already open
+     */
+    public void addNewFile(File file) throws JabRefException {
+        openFile(file);
+    }
+
+    /**
+     * Checks whether the file exists or if a new file should be opened.
+     * The file will be added to the set of journal abbreviation files.
+     * If the file also exists its abbreviations will be read and written
+     * to the abbreviations property.
+     *
+     * @param file to be opened
+     * @throws JabRefException if journal file with same path is already open
+     */
+    public void openFile(File file) throws JabRefException {
+        AbbreviationsFile abbreviationsFile = new AbbreviationsFile(file);
+        if (journalFiles.contains(abbreviationsFile)) {
```

Comment:
But there exists no Property for it which is needed in order to use the bindings.